File: SDS/ds/ds_H_5639.cpp
#include<iostream>

typedef struct Node{
    int value;
    Node *parent;
    Node *left;
    Node *right;
}Node;

void postorder(Node node)
{
    if(node == 0)
        return;
        
    postorder(node.left);
    postorder(node.right);
    printf("%d\n", node.value);
}

int main()
{
    Node root;
    int value;
    
    scanf("%d", &value);
    root = Node(value, 0, 0, 0);
    
    while(scanf("%d", &value) != -1)
    {
        Node node = root;
        while(1)
        {
            if(value > node.value)
            {
                if(node.right != 0)
                    node = node.right;
                else
                    node.right = Node(value, node, 0, 0);
                    break;
            }else{
                if(node.right != 0)
                    node = node.right;
                else
                    node.right = Node(value, node, 0, 0);
                    break;
            }
        }
    }
    
    postorder(root);
}

File: SDS/ds/ds_H_5639.py
from sys import stdin, stdout

class Node():
    def __init__(self, value, parent=None, left=None, right=None):
        self.value = value
        self.parent = parent
        self.left = left
        self.right = right

root = None

for value in stdin.readlines():
    value = int(value)
    if root == None:
        root = Node(value)
        continue
    
    node = root
    while True:
        if  value > node.value:
            if node.right != None:
                node = node.right
            else:
                node.right = Node(value, parent=node)
                break
        else:
            if node.left != None:
                node = node.left
            else:
                node.left = Node(value, parent=node)
                break
            
def postorder(node):
    if node == None:
        return
    
    postorder(node.left)
    postorder(node.right)
    stdout.write(str(node.value)+'\n')
    
postorder(root)
            